### ISSUE TITLE
LTX2.3 v2v IC-LoRA training loss plateaus at 0.3-0.4 and won't decrease further

### ISSUE THREAD

**Issue by segatecm** [AkaneTendo25/musubi-tuner#100]
### Description

I'm training LTX2.3's IC-LoRA with the `v2v` LoRA target preset, but the training loss consistently plateaus at around **0.3-0.4** and won't decrease further, regardless of training steps.

### Training Setup

- **Model:** LTX2.3
- **Training Mode:** IC-LoRA (`--ic_lora_strategy v2v`)
- **LoRA Target Preset:** `v2v` (`--lora_target_preset v2v`)
- **Network Dim:** 32
- **Network Alpha:** 32
- **Optimizer:** AdamW
- **Learning Rate:** 5e-4 (also tried 1e-4, 1e-3)
- **Weight Decay:** 0.0 (also tried 0.01)
- **Loss Type:** MSE (velocity target)
- **Timestep Sampling:** shifted logit normal (default)

### Problem Behavior

1. **Single data sample test:** Loss decreases from ~0.9 to ~0.35 in the first ~40 steps, then plateaus
2. <img width="329" height="313" alt="Image" src="https://github.com/user-attachments/assets/68954139-01fa-4070-a44c-68828221aa86" />
3. 
4. **Multiple data samples test:** Same behavior - loss plateaus at 0.3-0.4 range
5. **Tried without regularization:** Setting `--optimizer_args "weight_decay=0.0"` did not help
6. **Reference-target pairs:** Verified that reference videos are properly loaded and matched with target videos### Expected Behavior

Loss should continue to decrease as training progresses, especially with multiple training samples and disabled regularization.

### Questions

1. Is 0.3-0.4 a "natural floor" for v2v velocity loss in LTX2.3? If so, why?
2. Are there any known issues with v2v preset training that could cause this plateau?
3. What hyperparameters or configurations should I adjust to allow loss to continue decreasing?
4. Are there any validation/test metrics I should monitor to diagnose this further?

Any guidance or debugging suggestions would be greatly appreciated!

**Comment by AkaneTendo25** [comment]
I don't think v2v training loss is expected to go to zero, even on one sample. Flow-matching loss is an expectation over freshly drawn noise and timestep, not a fixed memorizable target, so a plateau at 0.3-0.4 by itself doesn't suggest training is unhealthy.

In the v2v IC-LoRA pipeline, the reference tokens are concatenated as clean conditioning tokens, get timestep `0`, and are excluded from the loss. The loss is only calculated against the target video tokens, with velocity target `noise-latents`. Since each step samples new noise and timestep/sigma, this is not the same as fitting one fixed deterministic target. Part of that plateau is irreducible: the velocity target contains the freshly drawn noise, and at low noise levels most of it simply cannot be predicted from a nearly clean input, no matter the capacity. On a single memorized sample it is recoverable in principle, but the exact solution amounts to amplifying a tiny input residual by 1/sigma, which a small LoRA delta on a frozen base cannot express.

In case you still suspect something is wrong with your parameters, check that the reference cache really contains the intended reference length. `--reference_frames` defaults to `1`, so if you did not cache with some specific value like `--reference_frames 25`, the training may effectively be image-reference conditioning rather than full v2v reference conditioning. 

For an actual sanity test, try restricting the timestep range. If loss falls much lower with a narrow range, the plateau is mostly from the broad stochastic velocity objective, not a broken training preset. For example, run a short overfit run with `--timestep_sampling uniform` and a narrow `--min_timestep` / `--max_timestep`. You can also test capacity separately: try a larger LoRA rank or `--lora_target_preset full`, and if that lowers the plateau, rank capacity is the bottleneck.

As for what to monitor: absolute velocity MSE is a weak signal on its own. Periodic samples (`--sample_prompts` with `--sample_every_n_steps`) or a small validation split (`[[validation_datasets]]` with `--validate_every_n_steps`) will tell you much more about whether the `v2v` conditioning is actually being learned than the loss curve.

**Comment by segatecm** [comment]
Thank you so much for your incredibly detailed and insightful response! 

I've learned so much from your explanation.Your work on musubi-tuner is absolutely amazing. I've been having so much fun exploring LTX-2.3 training with it! Thank you for creating such a robust tool and being so generous with your expertise.

**Comment by AkaneTendo25** [comment]
Thanks for the kind words! There are tons of things I don't know, and developing this fork is one of the ways I keep learning. Really happy to hear that others find it useful too! 😊
